scipy/optimize/minpack.py

```diff
@@ -411,7 +411,8 @@ def leastsq(func, x0, args=(), Dfun=None, full_output=0,
     m = shape[0]
 
     if n > m:
-        raise TypeError('Improper input: N=%s must not exceed M=%s' % (n, m))
+        raise TypeError('Improper input: func input vector length N=%s must '
+                        'not exceed func output vector length M=%s' % (n, m))
```

Comment:
Use f-strings and double quotes as much as possible. Also now we ask to put the space on the new line.
```suggestion
        raise TypeError(f"Improper input: func input vector length N={n} must"
                        f" not exceed func output vector length M={m}")
```